src/main/java/mekanism/client/render/tileentity/RenderTeleporter.java

```diff
@@ -12,16 +16,17 @@
 import net.minecraft.client.renderer.IRenderTypeBuffer;
 import net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher;
 import net.minecraft.profiler.IProfiler;
+import net.minecraft.util.Direction;
 
 @ParametersAreNonnullByDefault
 public class RenderTeleporter extends MekanismTileEntityRenderer<TileEntityTeleporter> {
 
-    private static Model3D EAST_WEST;
-    private static Model3D NORTH_SOUTH;
+    private static Map<Direction, Model3D> modelCache = new HashMap<>();
+    private static Map<Direction, Model3D> rotatedModelCache = new HashMap<>();
```

Comment:
These should be EnumMaps instead of HashMaps

Reply:
OK, i will change them.